core/src/engine/EngineMultiRobot.cc

```diff
@@ -2705,37 +2712,42 @@ namespace jiminy
                                         v);
 
             // Compute inertia matrix, adding rotor inertia.
-            pinocchio::crba(system.robot->pncModel_,
-                            system.robot->pncData_,
-                            q);
-            for (int32_t i = 1; i < system.robot->pncModel_.njoints; ++i)
-            {
-                // Only support inertia for 1DoF joints.
-                if (system.robot->pncModel_.joints[i].nv() == 1)
-                {
-                    int32_t const & jointIdx = system.robot->pncModel_.joints[i].idx_v();
-                    system.robot->pncData_.M(jointIdx, jointIdx) +=
-                            system.robot->pncModel_.rotorInertia[jointIdx];
-                }
-            }
+            pinocchio_overload::crba(system.robot->pncModel_,
+                                     system.robot->pncData_,
+                                     q);
 
             // Call forward dynamics.
-            return pinocchio::forwardDynamics(system.robot->pncModel_,
-                                              system.robot->pncData_,
-                                              q,
-                                              v,
-                                              uTotal,
-                                              system.robot->getConstraintsJacobian(),
-                                              system.robot->getConstraintsDrift(),
-                                              CONSTRAINT_INVERSION_DAMPING,
-                                              false);
+            a = pinocchio::forwardDynamics(system.robot->pncModel_,
+                                           system.robot->pncData_,
+                                           uTotal,
+                                           system.robot->getConstraintsJacobian(),
+                                           system.robot->getConstraintsDrift(),
+                                           CONSTRAINT_INVERSION_DAMPING);
         }
         else
         {
             // No kinematic constraint: run aba algorithm.
-            return pinocchio_overload::aba(
+            a = pinocchio_overload::aba(
                 system.robot->pncModel_, system.robot->pncData_, q, v, u, fext);
         }
+
+        // Compute the internal forces
+        pinocchio_overload::rnea(
+            system.robot->pncModel_, system.robot->pncData_, q, v, a);
+
+        // Using action-reaction law to compute the ground reaction force
+        system.robot->pncData_.f[0].setZero();
```

Comment:
I'm not sure to understand why you need this still. This would only have an impact if you run rnea an extra time, but isn't the first run enough ?

Reply:
This is not updated nor set to zero by rnea. Si it must be done manually.